server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/CacheConfigurationAdd.java

```diff
@@ -222,6 +222,9 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
         ModelNode resolvedValue;
         final String templateConfiguration = (resolvedValue = CacheConfigurationResource.CONFIGURATION.resolveModelAttribute(context, cacheModel)).isDefined() ? resolvedValue.asString() : null;
         final ModuleIdentifier moduleId = ModelNodes.asModuleIdentifier(CacheConfigurationResource.CACHE_MODULE.resolveModelAttribute(context, cacheModel));
+        if (moduleId != null) {
```

Comment:
Better solution: move the warning to https://github.com/infinispan/infinispan/blob/37e74f53bd426354e842dc9a41a9b96cdff8db65/server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLReader.java#L638 and also i18n it.